flask_monitoringdashboard/database/request.py

```diff
@@ -8,39 +8,28 @@
 from flask_monitoringdashboard.database import Request
 
 
-def get_latencies_in_timeframe(db_session, endpoint_id, start_date, end_date):
-    criterion = create_time_based_sample_criterion(start_date, end_date)
-
-    items = db_session.query(Request.duration).filter(Request.endpoint_id == endpoint_id, *criterion).all()
-
-    return [item.duration for item in items]
-
-
-def get_latencies_sample(db_session, endpoint_id, interval, sample_size=500):
-    criterion = create_time_based_sample_criterion(interval.start_date(), interval.end_date())
-
-    dialect = db_session.bind.dialect.name
-
-    if dialect in ['sqlite', 'mysql']:
-        order_by = func.random() if dialect == 'sqlite' else func.rand()
+def get_latencies_sample(session, endpoint_id, criterion, sample_size=500):
+    query = (
+        session.query(Request.duration).filter(Request.endpoint_id == endpoint_id,
+                                               *criterion)
+    )
+    # return random rows: See https://stackoverflow.com/a/60815
+    dialect = session.bind.dialect.name
 
-        items = db_session.query(Request.duration) \
-            .filter(Request.endpoint_id == endpoint_id, *criterion) \
-            .order_by(order_by) \
-            .limit(sample_size) \
-            .all()
+    if dialect == 'sqlite':
+        query = query.order_by(func.random())
+    elif dialect == 'mysql':
```

Comment:
what about other dialects? The SA answer states that it should be possible for Postgresql and Oracle DB

Reply:
This function produces a different query on different databases, and this is something that we should try to avoid as much as possible, as this will introduce unexpected errors. However, in this case, only the ordering is slightly different for different databases, and I don't see this as a problem.